src/awkward/_util.py

```diff
@@ -44,124 +72,357 @@ def is_sized_iterable(obj):
+            except Exception as err:
+                valuestr = f"repr-raised-{type(err).__name__}"
+
+        elif value is None or isinstance(value, (bool, int, float)):
+            try:
+                valuestr = repr(value)
+            except Exception as err:
+                valuestr = f"repr-raised-{type(err).__name__}"
+
+        elif isinstance(value, (str, bytes)):
+            try:
+                if len(value) < 60:
+                    valuestr = repr(value)
+                else:
+                    valuestr = repr(value[:57]) + "..."
+            except Exception as err:
+                valuestr = f"repr-raised-{type(err).__name__}"
+
+        elif isinstance(value, np.ndarray):
+            import numpy
+
+            if not numpy.__version__.startswith("1.13."):  # 'threshold' argument
+                prefix = f"{type(value).__module__}.{type(value).__name__}("
+                suffix = ")"
+                try:
+                    valuestr = numpy.array2string(
+                        value,
+                        max_line_width=width - len(prefix) - len(suffix),
+                        threshold=0,
+                    ).replace("\n", " ")
+                    valuestr = prefix + valuestr + suffix
+                except Exception as err:
+                    valuestr = f"array2string-raised-{type(err).__name__}"
+
+                if len(valuestr) > width and "..." in valuestr[:-1]:
+                    last = valuestr.rfind("...") + 3
+                    while last > width:
+                        last = valuestr[: last - 3].rfind("...") + 3
+                    valuestr = valuestr[:last]
+
+                if len(valuestr) > width:
+                    valuestr = valuestr[: width - 3] + "..."
+
+        elif isinstance(value, (Sequence, Mapping)) and len(value) < 10000:
+            valuestr = repr(value)
+            if len(valuestr) > width:
+                valuestr = valuestr[: width - 3] + "..."
+
+        if valuestr is None:
+            return f"{type(value).__name__}-instance"
+        else:
+            return valuestr
+
+
+class OperationErrorContext(ErrorContext):
+    def __init__(self, name, arguments):
+        string_arguments = {}
+        for key, value in arguments.items():
+            if isstr(key):
+                width = self._width - 8 - len(key) - 3
+            else:
+                width = self._width - 8
+
+            string_arguments[key] = self.format_argument(width, value)
+
+        super().__init__(
+            name=name,
+            arguments=string_arguments,
+            traceback=traceback.extract_stack(limit=3)[0],
+        )
+
+    @property
+    def name(self):
+        return self._kwargs["name"]
 
+    @property
+    def arguments(self):
+        return self._kwargs["arguments"]
 
-def deprecate(
-    message,
-    version,
-    date=None,
-    will_be="an error",
-    category=DeprecationWarning,
-    stacklevel=2,
-):
```

Comment:
This `deprecate` is a useful function that we'll likely want in version 2. It didn't get copied over into the v2 _utils.py because it hasn't been needed _yet_: we've been free to change behavior at will, since 2.0.0 hasn't been released.

But eventually, we'll need it, so just copy it verbatim from v1 _util.py into v2 _util.py.